ethabi/src/decoder.rs

```diff
@@ -202,9 +206,34 @@ fn decode_param(param: &ParamType, slices: &[[u8; 32]], offset: usize) -> Result
 				new_offset,
 			};
 
-			Ok(result)
-		}
-	}
+            Ok(result)
+        }
+        ParamType::Tuple(ref t) => {
+            let tuple_len_offset = if param.is_dynamic() {
+                let tuple_offset_slice = try!(peek(slices, offset));
+                (as_u32(tuple_offset_slice)? / 32) as usize
+            } else {
+                offset
+            };
+
+            let len = t.len();
+            let mut tokens = Vec::with_capacity(len);
+            let tuple_slices = &slices[tuple_len_offset..];
+            let mut new_offset = 0;
+            for i in 0..len {
+                let res = decode_param(&t[i], &tuple_slices, new_offset)?;
+                new_offset = new_offset + 1;
```

Comment:
Are all params here 32 bytes, so `new_offset + 1` is correct? Doesn't `decode_param` return a new offset?

Reply:
See the array decoding above, where it does `new_offset = res.new_offset`.